libvast/src/layout_version.cpp

```diff
@@ -0,0 +1,74 @@
+#include "vast/layout_version.hpp"
+
+#include "vast/detail/assert.hpp"
+#include "vast/error.hpp"
+#include "vast/io/read.hpp"
+#include "vast/io/write.hpp"
+#include "vast/logger.hpp"
+
+#include <fstream>
+#include <iterator>
+
+#include "caf/expected.hpp"
+
+namespace {
+
+const char* descriptions[]{
+  "invalid",
+  "v0",
+};
+
+static_assert(vast::layout_version{std::size(descriptions)}
+                == vast::layout_version::count,
+              "Mismatch between number of layout versions and descriptions");
+
+} // namespace
+
+namespace vast {
+
+// This must be in namespace `vast` so it can be found by overload resolution.
```

Comment:
That sounds like a bug, this function should be picked up via ADL.

Reply:
I think ADL works in the opposite direction? Ie. we could pick up additional functions in `vast::` if the argument is a `vast::layout_version`, but not additional functions from an unnamed namespace.

Reply:
I can't see any longer how it was because you force-pushed, but it seems to work now. Let's resolve this and discuss later when revisiting.